packages/palette/src/elements/ColumnGrid/ColumnGrid.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useMemo } from "react"
+import styled from "styled-components"
+import { gridColumn, GridColumnProps } from "styled-system"
+import { Box, BoxProps } from "../Box"
+import { CSSGrid, CSSGridProps } from "../CSSGrid"
+
+/** ColumnGrid implements `Box` and the common grid properties */
+export type ColumnGridProps = CSSGridProps
+
+/**
+ * A 12-column fluid grid. Collapses to a single column at mobile breakpoints.
+ */
+export const ColumnGrid = styled(CSSGrid)``
+
+ColumnGrid.defaultProps = {
+  gridColumnGap: 2,
+  gridRowGap: 2,
+  gridTemplateColumns: "repeat(12, 1fr)",
+}
+
+type ColumnSpan = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12
+type ColumnStart = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12
+
+interface ColSpanProps {
+  span?: ColumnSpan | ColumnSpan[]
+  start?: ColumnStart | ColumnStart[]
+}
+
+type __ColumnProps__ = ColSpanProps & GridColumnProps & BoxProps
+
+const __Column__ = styled(Box)<__ColumnProps__>`
+  ${gridColumn}
+`
+
+/**
+ * Builds a value for `gridColumn` based on the assumptions laid out
+ * by our 12-column grid.
+ */
+export const calculateGridColumn = ({ span, start }: ColSpanProps) => {
+  const spans = (() => {
```

Comment:
non-blocking: I'm curious about the use of IIFE's here. Is there a need for them, or is it a stylistic choice to avoid creating a function elsewhere?

Reply:
Yeah, initially. Kind of the equivalent of doing this in Ruby:

```ruby
foo = begin
  ...
end
```

Though with the changes I just made, this got complex enough for it to actually be extracted.